authentication/serializers.py

```diff
@@ -95,7 +95,13 @@ def _authenticate(self, flow):
                     authentication_flow
                     and authentication_flow == SocialAuthState.FLOW_REGISTER
                 ):
-                    email = partial.data.get("kwargs").get("details").get("email")
+                    email = (
+                        partial.data.get("kwargs")
+                        .get("details")
+                        .get("email")
+                        .strip()
+                        .lower()
```

Comment:
If we have started to make the email lower in the pipeline step, Do we still need to again this lower function here ?

Reply:
If we need it here, we should add test for this as well.

Reply:
As mentioned in the above comment https://github.com/mitodl/mitxpro/pull/2642#discussion_r1192248397. This is where the `InvalidEmail` exception is being handled. So adding `.strip().lower()` is required here to check if the user exists in the `InvalidEmail` Exception.

Reply:
I've removed `.strip().lower()` from here and fixed to query to check the user with case insensitive email